api/spec/swagger_definitions/comments.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+require_relative 'base_types'
+
+module Comments
+  class << self
+
+    def request_create
+      Type.submit_wrapper(
+        Type.object({
+          body: Type.string,
+          parent_id: Type.id,
+          deleted: Type.boolean
+        })
+      )
+    end
+
+    def request_update
+      Type.submit_wrapper(
+        Type.object({
+          body: Type.string,
+          deleted: Type.boolean
+        })
+      )
+    end
+
+    def response
+      Type.object({
+        data: Type.object({
+          id: Type.id,
+          type: Type.string,
+          attributes: Type.object({
+            body: Type.string,
+            parent_id: Type.id,
+            created_at: Type.date_time,
+            flags_count: Type.integer,
+            flagged: Type.boolean,
+            abilities: Type.object( Type.crud ), # TODO check that crud is the right abilities type
+            deleted: Type.boolean,
+            children_count: Type.integer,
+            sort_order: Type.integer,
+            author_created: Type.boolean,
+          }),
+          relationships: Type.object({
+            creator: Type.reference( '#/definitions/Maker' ) # TODO check that this is indeed a maker type
```

Comment:
The `creator` relationship will always (or almost always) be a User, not a Maker.

Reply:
Is that the case as well for the creators on the `project` type?

```
    def request_create
      Type.object({
        data: Type.object({
          attributes: Type.object(request_create_attributes, { required: required_create_attributes }),
          relationships: Type.object({
            creators: Type.reference('#/definitions/Maker'),
            collaborators: Type.reference('#/definitions/Maker'),
            contributors: Type.reference('#/definitions/Maker'),
            subjects: Type.reference('#/definitions/Subject')
          })
        })
      })
    end
```

Reply:
No. I'm referring specifically to the `creator` relationship. I believe that ever time you see a belongs_to `creator` relationship, it is a relationship to a single user object. In the case of the project, which has_many creators, it is a reference to makers.

On projects, creators is an array of makers. contributors is an array of makers, and collaborators is an array of collaborators, which is a different model from maker. See app/models/concerns/collaborative.rb for some more details about what's happening here.